### PR TITLE
API all - admin

### LINKED ISSUE #11: API all - admin
GET all hotel owners
GET all users
PUT (accept) hotel owner
POST/DELETE remove hotel owner by userName or email
POST/DELETE remove user by userName or email

all of these need auth

### PR DESCRIPTION
**Pull request by ibednorz**

### CODE REVIEW

File: routes/admin.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+router.put('/owner/accept/:email', async (req, res) => {
+    const email = req.params.email
+
+    try {
+        await User.updateOne({email: email}, {role: 'hotelOwner'})
+
+        res.status(200).send('Done')
+    } catch (err) {
+        console.log(err)
+        res.status(500).send('Something went wrong')
```
Comment (GoraJakub) [inline routes/admin.js]: Could you change errors to 'throw new ApiError(code,message)'?

**Review by GoraJakub — CHANGES_REQUESTED** [review]
Please redo all errors

File: routes/admin.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+router.get('/', async (req, res) => {
+  const users = await User.find()
+
+  res.send(users)
+})
+
+router.get('/owners', async (req, res) => {
+    const owners = await User.find(
+        { "role": /hotelOwner/i }
```
Comment (bartoszbialecki) [inline routes/admin.js]: hotelOwner should not be used like this, we have roles.js in models directory so use it

File: routes/admin.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        console.log(err)
+        res.status(500).send('Something went wrong')
+    }
+})
+
+router.delete('/owner/:email', async (req, res) => {
+    const email = req.params.email
+    
+    try {
+        const id = await User.find({ email: email, "role": /hotelOwner/i }, '_id')
```
Comment (bartoszbialecki) [inline routes/admin.js]: I think you can find and delete in one query

**Review by bartoszbialecki — COMMENTED** [review]
The same comment as in Kuba's PR, I think we should use everywhere json, so try res.json instead of res.send.
Your endpoints are not protected, everyone can call them, not only admin user

File: routes/admin.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        const id = await User.find({ email: email, "role": /user/i }, '_id')
+        if(id.length === 0){
+            res.status(400).send('Wrong email')
+        }
+        await User.findByIdAndDelete(id)
+
+        res.status(200).send(id)
+    } catch (err) {
+        console.log(err)
+        res.status(500).send('Something went wrong')
```
Comment (pdybowski) [inline routes/admin.js]: Send ({ message: 'something went wrong' }) 
Or use .json 
Whatever

File: routes/admin.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    console.log(err)
+    throw new ApiError(500, 'Something went wrong')
+  }
+})
+
+router.delete('/owner/:email', isAdmin, async (req, res) => {
+  const email = req.params.email
+
+  try {
+    const id = await User.findOneAndDelete({
```
Comment (pdybowski) [inline routes/admin.js]: check and fix

File: routes/admin.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  res.json(users)
+})
+
+router.get('/owners', isAdmin, async (req, res) => {
+  const owners = await User.find({ role: HOTEL_OWNER_ROLE })
+
+  res.json(owners)
+})
+
+router.put('/owner/accept/:email', isAdmin, async (req, res) => {
```
Comment (pdybowski) [inline routes/admin.js]: move isAdmin middleware to startu/routes

**Pull request merged**